### PR TITLE
Fix Chartfactory

### PR DESCRIPTION
**Pull request by karimkawambwa**
## Description

Fix issues where prop spreading would override computed properties.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/factory/ChartFactory.js [hunk 1]
```diff
@@ -251,7 +247,8 @@ function ChartFactory({
         domainPadding.x[0] +
         domainPadding.x[1];
       const height = visualProps.height || theme.bar.height;
-      const computedWidth = horizontal ? height : computedSize;
+      const computedWidth =
+        horizontal || computedSize < height ? height : computedSize;
```
Comment (kilemensi) [inline src/factory/ChartFactory.js]: I think:
1. `horizontal` if requested, _or_
2. `horizontal` if `computedSize > width` i.e. it won't fit the `width`.
Screen height is not really limited but width is.

PS: We should do this for `grouped_column` as well (and every other place we're computing the size).

File: stories/factory.stories.js [hunk 2]
```diff
@@ -129,31 +130,30 @@ storiesOf('HURUmap UI|Charts Factory/ChartFactory', module)
...
+        data={data.map((_, index) => {
+          const y = rand();
+          return {
+            tooltip: `Data ${index}`,
+            x: `Data ${index}`,
+            y
+          };
+        })}
+        refrenceData={refrence.map((_, index) => ({
+          label: 'Refrence',
```
Comment (kilemensi) [inline stories/factory.stories.js]: ~Refrence~Reference

File: src/factory/ChartFactory.js (hunk 1 above)
Comment (KhadijaMahanga) [reply src/factory/ChartFactory.js]: @kilemensi  spot on :100:

File: src/factory/ChartFactory.js [hunk 3]
```diff
@@ -188,32 +185,41 @@ function ChartFactory({
...
+      };
       const computedSize =
         primaryData.length * barCount * offset +
         domainPadding.x[0] +
         domainPadding.x[1];
       const height = visualProps.height || theme.bar.height;
-      const computedWidth = horizontal ? height : computedSize;
+      const width = visualProps.width || theme.bar.width;
+      const computedWidth =
+        horizontal || computedSize > width ? width : computedSize;
```
Comment (kilemensi) [inline src/factory/ChartFactory.js]: We need to do the same check for `computedHeight` as well .... otherwise it doesn't work

**Pull request merged**